### PR TITLE
feat(starred-count): show starred count

### LINKED ISSUE #129: Show starred repositories
It was nice to show them on the profile page after the repositories :star: 

image

### PR DESCRIPTION
**Pull request by adrianhartanto0**
This is to close #129. I followed the suggestion made by @RolfKoenders, and used `XMLHttpRequest` instead of `fetch` to retrieve the star count. 

I have also included a dutch and french translation for the "star" word using Google translate. @RolfKoenders @machour, could you please verify that the translated word is correct?

### CODE REVIEW

File: src/api/index.js [hunk 1]
```diff
@@ -310,6 +310,48 @@ export async function fetchForkRepo(owner, repo, accessToken) {
   return response;
 }
 
+export const requestStarCount = owner => {
+  const ENDPOINT = `https://api.github.com/users/${owner}/starred?per_page=1`;
+
+  const requestPromise = new Promise((resolve, reject) => {
+    const req = new XMLHttpRequest();
```
Comment (lex111) [inline src/api/index.js]: You can use `fetch` instead, quick example:

```js
fetch('https://api.github.com/users/lex111/starred?per_page=1').then(function (response) {
  console.log(response.headers.get('Link'));
});
```

File: src/components/user-profile.component.js [hunk 2]
```diff
@@ -141,6 +143,15 @@ export const UserProfile = ({
         </Text>
       </TouchableOpacity>
 
+      <TouchableOpacity style={styles.unit}>
+        <Text style={styles.unitNumber}>
+          {starCount}
+        </Text>
+        <Text style={styles.unitText}>
+          {translate('common.star', language)}
```
Comment (lex111) [inline src/components/user-profile.component.js]: It is better to use the plural (`stars`), by analogy with other (`repositories`, `followers`).

File: src/locale/languages/fr.js [hunk 3]
```diff
@@ -288,6 +288,7 @@ export const fr = {
   },
   common: {
     bio: 'BIO',
+    star: 'étoiles',
```
Comment (lex111) [inline src/locale/languages/fr.js]: If look at the key `starsTitle`, then the stars are translated as "favoris".

File: src/locale/languages/nl.js [hunk 4]
```diff
@@ -282,6 +284,7 @@ export const nl = {
   },
   common: {
     bio: 'BIO',
+    star: 'Sterren',
```
Comment (lex111) [inline src/locale/languages/nl.js]: And here it will be simple - `Stars` (again, if look at the key `starsTitle`)

File: src/user/user.type.js [hunk 5]
```diff
@@ -8,3 +8,4 @@ export const GET_FOLLOWERS = createActionSet('GET_FOLLOWERS');
 export const GET_FOLLOWING = createActionSet('GET_FOLLOWING');
 export const SEARCH_USER_REPOS = createActionSet('SEARCH_USER_REPOS');
 export const CHANGE_FOLLOW_STATUS = createActionSet('CHANGE_FOLLOW_STATUS');
+export const GET_STAR_COUNT = createActionSet('CHANGE_FOLLOW_STATUS');
```
Comment (lex111) [inline src/user/user.type.js]: Must be `GET_STAR_COUNT`

**Comment by adrianhartanto0** [comment]
5ab1671 reduce the left and right margin of the badge in `user-profile-component.js` to make it one line instead of multiline. 

original-margin

new-margin

File: src/user/user.reducer.js [hunk 6]
```diff
@@ -63,6 +65,24 @@ export const userReducer = (state = initialState, action = {}) => {
         error: action.payload,
         isPendingOrgs: false,
       };
+    case GET_STAR_COUNT.PENDING:
+      return {
+        ...state,
+        starCount: ' ',
```
Comment (lex111) [inline src/user/user.reducer.js]: Probably it is possible to remove, after all in `initialStore` already there is this key?

File: src/locale/languages/fr.js (hunk 3 above)
Comment (machour) [inline src/locale/languages/fr.js]: yes, it should be **favoris**. Merci beaucoup Mr @lex111 !

File: src/locale/languages/nl.js [hunk 7]
```diff
@@ -282,6 +284,7 @@ export const nl = {
   },
   common: {
     bio: 'BIO',
+    stars: 'Stars',
```
Comment (RolfKoenders) [inline src/locale/languages/nl.js]: Dutch should be: 'Sterren'

Comment (adrianhartanto0) [inline src/locale/languages/nl.js]: Thanks for your help! Will change the translation

**Comment by housseindjirdeh** [comment]
Overall this looks so solid :O :O Thanks so much @adrianhartanto0, there are literally just the few things @RolfKoenders and @lex111 noticed that I can see will need changing but otherwise this should be solid to merge :tada:

File: src/user/user.reducer.js (hunk 6 above)
Comment (adrianhartanto0) [reply src/user/user.reducer.js]: @lex111 I have decided to remove it from `initialStore`, and only add to it when `GET_STAR_COUNT.PENDING` and `GET_AUTH_STAR_COUNT.PENDING` are respectively dispatched.

Comment (lex111) [reply src/user/user.reducer.js]: Well, I would still put it in `initialStore`, and then it could only be added to `GET_STAR_COUNT.SUCCESS`, but I'm ok.

**Pull request merged**